### PR TITLE
Add Module Marketplace

### PR DESCRIPTION
**Pull request by tuxudo**
An admin page that details installed modules and available widgets

### CODE REVIEW

**Comment by joncrain** [comment]
My initial install did not pull in all database columns for some reason. After I reran my dev setup script, it started working.
image

Also the enabled modules does not seem to be working for me. When I have nothing in my .env for "MODULES", then all show up as enabled. After I added a few in the .env, all show up as no.

Also, the path for core modules has an added `/` `/munkireport_dev//vendor/munkireport/supported_os`

**Comment by tuxudo** [comment]
The double slash is how MR returns the data, I'll correct it so it doesn't show. Not sure why your modules are acting that way. Will look into that more tomorrow.

**Comment by joncrain** [comment]
If I surround my module list with `'`, it works, `"` does not work. 

Also, how are you pulling date downloaded? It looks to be the same as date updated.

**Comment by tuxudo** [comment]
If your modules are up to date, then downloaded and uploaded will be the same. May change that to the timestamp of the model file itself. 

The `'` and `"` is in .env? It's just going off of what MR gives it

**Comment by joncrain** [comment]
Yeah, I'm not sure what's going on. It hasn't been consistent for me. I'll check it again in the morning though.

**Comment by tuxudo** [comment]
Fixed the double slash, first loading of the MM on a new install, fixed some deprecation warnings, fixed enabled logic

File: app/controllers/Module_marketplace.php [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+use Symfony\Component\Yaml\Yaml;
+use munkireport\models\Cache;
+
+class Module_marketplace extends Controller
+{
+    private $moduleMarketplace;
+
+    public function __construct()
+    {
+        $this->moduleMarketplace = getMrModuleObj();
```
Comment (bochoven) [inline app/controllers/Module_marketplace.php:16]: // Check authorization
        $this->authorized() || jsonError('Authenticate first', 403);
        $this->authorized('global') || jsonError('You need to be admin', 403);

Comment (tuxudo) [reply app/controllers/Module_marketplace.php:22]: Added

**Comment by tuxudo** [comment]
Not yet ready to merge, working on removing the need for the database part of it

**Comment by tuxudo** [comment]
PR ready to be reviewed and merged.

**Comment by bochoven** [comment]
If the database is not needed anymore, the migration is also not necessary?

**Comment by tuxudo** [comment]
Database elements and migration removed

**Pull request merged**